Source/Core/DolphinQt2/Settings/AudioPane.cpp

```diff
@@ -0,0 +1,236 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Settings/AudioPane.h"
+
+#include <QCheckBox>
+#include <QComboBox>
+#include <QFormLayout>
+#include <QGridLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QRadioButton>
+#include <QSlider>
+#include <QSpacerItem>
+#include <QSpinBox>
+#include <QVBoxLayout>
+
+#include "AudioCommon/AudioCommon.h"
+#include "Core/ConfigManager.h"
+#include "DolphinQt2/Config/SettingsWindow.h"
+#include "DolphinQt2/Settings.h"
+
+AudioPane::AudioPane()
+{
+  CreateWidgets();
+  LoadSettings();
+  ConnectWidgets();
+
+  connect(&Settings::Instance(), &Settings::VolumeChanged, this, &AudioPane::OnVolumeChanged);
+}
+
+void AudioPane::CreateWidgets()
+{
+  auto* dsp_box = new QGroupBox(tr("DSP Emulator Engine"));
+  auto* dsp_layout = new QVBoxLayout;
+  auto* dsp_layout_combo = new QVBoxLayout;
```

Comment:
I don't understand this. There's a VBoxLayout which contains two VBoxLayouts, of which the first one has widgets and the second one is empty? I'm surprised that Qt interprets this to mean "vertically center the first child VBox". A clearer way to do this is:

```c++
 auto* dsp_layout = new QVBoxLayout;
dsp_box->setLayout(dsp_layout);
dsp_layout->addStretch(1);
dsp_layout->addWidget(m_dsp_hle);
dsp_layout->addWidget(m_dsp_lle);
dsp_layout->addWidget(m_dsp_interpreter);
dsp_layout->addStretch(1);
```

From the UI perspective, is vertical centering desired here? On Windows, according to @Starsam80, the radio buttons are aligned to the top-left of the box.